File: lib/demo/random_movement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const ConsoleOut_1 = require("../display/ConsoleOut");
const SnakeGame_1 = require("../SnakeGame");
const util_1 = require("../util");
const display = new ConsoleOut_1.default();
const game = new SnakeGame_1.default(8, 8);
const game_immediate = setInterval(() => {
    game.tick();
    if (Math.random() >= .5) {
        const cmd = [
            SnakeGame_1.Command.UP,
            SnakeGame_1.Command.RIGHT,
            SnakeGame_1.Command.DOWN,
            SnakeGame_1.Command.LEFT,
        ][util_1.rand(0, 3)];
        game.invoke(cmd);
    }
    display.show(game.screen());
}, 1000);

File: random_movement.py
from time import sleep

from src.Display import Display
from src.display.ConsoleOut import ConsoleOut 
from src.SnakeGame import SnakeGame, Command
from src.util import rand

if __name__ == '__main__':
  display = ConsoleOut()
  game = SnakeGame(None, 8, 8)
  try:
    for i in range(0, 5):
      game.tick()

      if rand(0, 1) == 0:
        cmd = [
          Command.UP,
          Command.RIGHT,
          Command.DOWN,
          Command.LEFT,
        ][rand(0, 3)]

        game.invoke(cmd)

      display.show(game.screen())
      sleep(1)
  finally:
    display.tearDown()
